### ISSUE TITLE
How to handle json data types?

### ISSUE THREAD

**Issue by noah-integrityinspired** [byme8/ZeroQL#52]
**Is your feature request related to a problem? Please describe.**
My graphQL has a JSON* field that I would like to handle; but I cannot find a way to do this with ZeroQL; haven't had much luck with custom serializers yet. 

*(Technically JSONB; we are using Hasura w/ Postgres)

For example I'd like to turn this data:
`{ "name": "john", "address": { "street": 1234, "zip": 45567, } }`
and put it into a shape like:
```cs
class Person 
{
  public string Name {get; set;}
  public jsonb Address {get; set;}
}
```

So far we've tried the code below, but we haven't had any success. Was wondering if you had any examples you could share or could help us figure out what we are missing.

Out of the box the code generation creates this scalar in the generated code. 
```cs
public sealed record jsonb : ZeroQLScalar
{
    public jsonb()
    {
    }

    public jsonb(string value)
    {
        Value = value;
    }

    public static implicit operator jsonb(string value) => new jsonb(value);
    public static implicit operator string (jsonb scalar) => scalar.Value;
}
```
But we are unable to use it as it chokes on  the opening curly brace ('StartObject') during deserialization.  

So we have tried to create a custom converter, but we are unable to get any of the Console.WriteLine statements to occur; suggesting it's not registering correctly with ZeroQL?
```cs
public class JsonBConverter : JsonConverter<jsonb>
{
    public override jsonb? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        Console.WriteLine("Beginning read...");
        throw new NotImplementedException();
    }

    public override void Write(Utf8JsonWriter writer, jsonb value, JsonSerializerOptions options)
    {
        Console.WriteLine("Beginning write...");
        throw new NotImplementedException();
    }
}

public static class Module
{
    [ModuleInitializer]
    public static void Initialize()
    {
        ZeroQLJsonOptions.Configure(o => o.Converters.Add(new JsonBConverter()));
    }
}
```

**Comment by byme8** [comment]
Have you seen https://github.com/byme8/ZeroQL/wiki/User-scalars? 

If you want to create your own serialization for a custom scalar, you need to do the next steps:
- create "container" for it 
``` csharp
// something like that
namespace MyCustom;

public class jsonb : JsonElement
{
}
```
- define serializer you did it
- tell zeroql generator to replace "stub scalar" with your custom one  ``--scalars jsonb=MyCustom.jsonb ``

Looks like you skipped the first and the last step.

**Comment by noah-integrityinspired** [comment]
Thank you for your patience! We tired doing the first and last step you mentioned, but it did not work in our earlier attempts

But we've figured it out.
To future readers: if you are using the a `zeroql.json` file, try generating without the the json and use the `--scalars` arg; it may expose the issue we were dealing with. 

What we experienced:

The project we were working in had only installed `ZeroQL` v4.1. However the project somehow had access to the ZeroQL cli, but only v4.0. And because we are using `zeroql.json` file, it did not give us any errors when we went to generate.

When we tried generating without the json file, using the --scalars arg, it gave us an error:
```
Unrecognized option(s):
--scalars
```
And told us it it was on v4.0.

After installing the `ZeroQL.CLI` tool, the generate command now works correctly and uses the custom class in the generate code.
